src/mlpack/methods/ann/layer_names.hpp

```diff
@@ -307,6 +307,12 @@ class LayerNameVisitor : public boost::static_visitor<std::string>
     return "unsupported";
   }
 
+  //! Overload function call.
+  std::string operator()(MoreTypes* layer) const
```

Comment:
It should actually be `MoreTypes` not `MoreTypes*` :)

Reply:
Oh, I was a bit confused regarding that part and chose to go with ```MoreTypes*```. I don't have the latest build on this setup so I couldn't compile-check it either. Thanks for pointing out, I have fixed it now :)